lib/utils/file.js

```diff
@@ -9,11 +9,29 @@
  * Issue: https://github.com/kevva/executable/issues/9
  * PR: https://github.com/kevva/executable/pull/10
  */
+import fs from 'fs';
+import {arrBuf2Base64} from './array';
 
-export default function isExecutable(fileStat) {
+export function isExecutable(fileStat) {
   if (process.platform === 'win32') {
     return true;
   }
 
   return Boolean(fileStat.mode & 0o0001 || fileStat.mode & 0o0010 || fileStat.mode & 0o0100);
 }
+
+export const getFileDataAsync = soundFileLocation => {
```

Comment:
The desired file type would be that of a sound file which is why it is so specific. It might even be worth evaluating the extension or file type somehow to potentially prevent reading a file we know we cant use for a bell sound? If desired, we can use a more generic name here.

Reply:
I think this function is more generic than just for sound files, so either rename it to `filePath`, and implement the sound checks in another function, **or** rename this function to `getSoundDataAsync`?

Reply:
I think that we need to make this function less specific:

We have to export a `getBase64File` function or something like that. And merge `arrBuf2Base64` (updated to use Buffers) code into it.
This function will be responsible to load a filePath asynchronously and returns a base64 string from its content.
Adding `base64,` prefix should be done after using this function because this is really specific and will prevent to reuse this function.

I think that we don't need to make any filetype verification.

Reply:
I see what you all mean now and agree that this should be more generic for better reusability. Ill get this updated with the feedback in mind. @chabou Thank you, btw, for the feedback on using the Buffer class to create the base64 string. I felt that there should be a better way of achieving the same result but decided to begin with a similar implementation to the one being used in the `SoundManager.ts` file in the xterm.js project to mitigate bugs caused by mismatched methods and results.

Reply:
There is a huge difference between xtermjs SoundManager and your function: yours is executed on Electron MainProcess (ie nodejs, not a browser). And Buffer class is only available in NodeJs.